demos/_unit-test/stream.php

```diff
@@ -10,15 +10,23 @@
 /** @var \Atk4\Ui\App $app */
 require_once __DIR__ . '/../init-app.php';
 
+/**
+ * Size in Mb of generated download file.
+ * This can also be passed as GET['size_mb'] parameter.
+ *
+ * @var int
+ */
+$size_mb = (int) ($_GET['size_mb'] ?? 64);
+
 $button = Button::addTo($app, ['Download', 'class.atk-test' => true]);
-$button->on('click', function () use ($app) {
+$button->on('click', function () use ($app, $size_mb) {
     // Generate big data and write it to a temporary file
     $pattern = str_repeat('0123456789ABCDEF', 65536); // 1Mb
-    $chunks = 64; // 1 chunk = 1Mb, 1024 chunks = 1Gb etc
+    $total_size = strlen($pattern) * $size_mb;
     $tempFile = tempnam(sys_get_temp_dir(), 'test.txt');
```

Comment:
I would like to see here a dynamic stream emitter via callback /wo intermediate temporary file.

On the DemosTest side, there should be a reader which will read the stream in parts, compare it, and discard immediatelly.

Reply:
implemented